umbra-js/src/classes/Umbra.ts

```diff
@@ -235,6 +256,119 @@ export class Umbra {
+          throw new Error(`Insufficient balance to complete transfer. ${details}`);
+        }
+      }
+    }
+
+    const txSigner = this.getConnectedSigner(signer); // signer input validated
+
+    // Get toll amount from contract
+    const toll = await this.umbraContract.toll();
+
+    // Parse provided overrides
+    const localOverrides = { ...overrides }; // avoid mutating the object passed in
+    const advanced = localOverrides?.advanced || false;
+    const supportPubKey = localOverrides?.supportPubKey || false;
+    const supportTxHash = localOverrides?.supportTxHash || false;
+    const lookupOverrides = { advanced, supportPubKey, supportTxHash };
+
+    delete localOverrides.advanced;
+    delete localOverrides.supportPubKey;
+    delete localOverrides.supportTxHash;
+
+    let sendData: SendData[] = [];
+    let valueAmount: BigNumber = BigNumber.from('0');
+    let stealthKeyPairs: KeyPair[] = [];
+
+    for (let i = 0; i < batch.length; i++) {
+      // Lookup recipient's public key
+      const { spendingPublicKey, viewingPublicKey } = await lookupRecipient(
+        batch[i].address,
+        this.provider,
+        lookupOverrides
+      );
+      if (!spendingPublicKey || !viewingPublicKey) {
+        throw new Error(`Could not retrieve public keys for recipient ID ${batch[i].address}`);
+      }
+
+      const token = batch[i].token;
+      const amount = batch[i].amount;
+
+      const spendingKeyPair = new KeyPair(spendingPublicKey);
+      const viewingKeyPair = new KeyPair(viewingPublicKey);
+
+      // Generate random number
+      const randomNumber = new RandomNumber();
+
+      // Encrypt random number with recipient's public key
+      const encrypted = viewingKeyPair.encrypt(randomNumber);
+
+      // Get x,y coordinates of ephemeral private key
+      const { pubKeyXCoordinate } = KeyPair.compressPublicKey(encrypted.ephemeralPublicKey);
+
+      // Compute stealth address
+      const stealthKeyPair = spendingKeyPair.mulPublicKey(randomNumber);
+      stealthKeyPairs.push(stealthKeyPair);
+
+      // Ensure that the stealthKeyPair's address is not on the block list
+      if (blockedStealthAddresses.includes(stealthKeyPair.address)) throw new Error('Invalid stealth address');
+
+      // Sum valueAmount
+      if (isEth(token)) {
+        valueAmount = valueAmount.add(toll.add(amount));
+      } else {
+        valueAmount = toll.add(valueAmount);
+      }
+      sendData.push({
+        receiver: stealthKeyPair.address,
+        tokenAddr: token,
+        amount: amount,
+        pkx: pubKeyXCoordinate,
+        ciphertext: encrypted.ciphertext,
+      });
+    }
+
+    // Sort by token addresses
+    const sortedData: SendData[] = sendData.sort((n1, n2) => {
+      return n1.tokenAddr.localeCompare(n2.tokenAddr);
+    });
+
+    // Send transaction
+    let tx: ContractTransaction;
+    const txOverrides = { ...localOverrides, value: valueAmount, gasLimit: 5000000 };
```

Comment:
Why do we hardcode a gas limit here? We should be able to remove that
```suggestion
    const txOverrides = { ...localOverrides, value: valueAmount };
```

Reply:
I was getting gas errors when testing along with the frontend, not sure why. But removed it for now!

Reply:
Do you still get errors now that it's removed? If so what is the specific error / how can I reproduce?

Reply:
It seems to be working without the manual gas limit now, on branch `frontend-batch-send-demo`

Reply:
Ok going to mark this resolved then